fpm/fpm.toml

```
@@ -10,6 +10,9 @@ copyright = "2020 fpm contributors"
 git = "https://github.com/toml-f/toml-f"
 tag = "v0.2"
 
+[dependencies.M_CLI2]
+git = "https://github.com/urbanjost/M_CLI2.git"
```

```suggestion
git = "https://github.com/urbanjost/M_CLI2.git"
commit = "a177b0077819571815fa6a8da6980bcb45443858"
```

We need to pin our dependencies to a specific commit or release. Here I've chosen the most recent from M_CLI2.

---

The keyword commit= works with the Haskell fpm but when I try that with the toml-f is says commit= is not allowed. Otherwise I agree with this.  Should that be a separate issue?  Or should I add a tag? I have no problem with just including the M_CLI2.f90 file but I was following the (really nice) toml-f implementation

---

Apologies, this is my mistake. It should be `rev` instead of `commit`.
```suggestion
git = "https://github.com/urbanjost/M_CLI2.git"
rev = "a177b0077819571815fa6a8da6980bcb45443858"
```

---

It is actually more typical to do all the command line parsing in one place with M_CLI2, but the arguments can be called from anywhere. Since all the model settings and such were being set and used in fpm I just parsed and provided help text in the command_line procedure and retrieved them in the cmd routines where they were used. It is trivial to move the calls to get_args(3f) back into command_line; just not sure whether it means adding options to the settings or to the calls. I am looking at the code trying to see what the model is there.